### PR TITLE
Implement profile cpu crd

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
This PR implements CRD support for the "profile cpu" gadget and removes the bcck8s package that is not needed anymore.
CO-RE support is not implemented here but it can be done separately. 

Related https://github.com/kinvolk/inspektor-gadget/issues/533

---

https://github.com/kinvolk/inspektor-gadget/pull/648 should be reviewed first.

### CODE REVIEW

File: pkg/gadgets/profile/gadget.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	"os/exec"
+	"syscall"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/apis/gadget/v1alpha1"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/profile/types"
+)
+
+type Trace struct {
+	resolver gadgets.Resolver
```
Comment (blanquicet) [inline pkg/gadgets/profile/gadget.go:29]: I think it is not needed.

File: pkg/gadgets/profile/gadget.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+func NewFactory() gadgets.TraceFactory {
+	return &TraceFactory{
+		BaseFactory: gadgets.BaseFactory{DeleteTrace: deleteTrace},
+	}
+}
+
+func (f *TraceFactory) Description() string {
+	return `Profile CPU usage by sampling stack trace TODO: Complete`
```
Comment (blanquicet) [inline pkg/gadgets/profile/gadget.go:48]: Just adding a comment here so that we do not forget to complete it.

**Review by eiffel-fl — APPROVED** [review]
A PR with a negative patch :partying_face:!
I tested it and it works as excepted, so thank you for it!

**Review by blanquicet — APPROVED** [review]
LGTM!

**Pull request merged**